pkg/cli/codearea.go

```diff
@@ -196,6 +200,56 @@ func (w *codeArea) handlePasteSetting(start bool) bool {
 	return true
 }
 
+// Try to expand a non-word abbreviation.
+func (w *codeArea) expandAbbr() {
+	var abbr, full string
+	// Try to expand an abbreviation, preferring the longest one
+	w.Abbreviations(func(a, f string) bool {
+		if strings.HasSuffix(w.inserts, a) && len(a) > len(abbr) {
+			abbr, full = a, f
+		}
+		return false // check the next abbr
+	})
+	if len(abbr) > 0 {
+		c := &w.State.Buffer
+		*c = CodeBuffer{
+			Content: c.Content[:c.Dot-len(abbr)] + full + c.Content[c.Dot:],
+			Dot:     c.Dot - len(abbr) + len(full),
+		}
+		w.resetInserts()
+	}
+}
+
+// Try to expand a small word abbreviation.
+func (w *codeArea) expandSmallWordAbbr() {
+	w.StateMutex.Lock()
+	defer w.StateMutex.Unlock()
+
+	ilen := len(w.inserts)
+	w.WordAbbreviations(func(a, f string) bool {
+		alen := len(a)
+		if alen > ilen { // skip abbr since it can't match the insert buffer
+			return false // check the next abbreviation
+		}
+		if ilen > alen {
+			rune, _ := utf8.DecodeLastRuneInString(w.inserts[:ilen-alen])
```

Comment:
IIUC this only checks the *left border* of the abbreviation. The right border should be checked too, using the to-be-inserted character.

Reply:
That's done before calling this function. However, your comment caused me to realize there was a corner case this doesn't handle correctly. Consider the abbreviation `gc` => `git commit` and the user types `x`, ` `, [backspace], `g`, `c`, ` `. This implementation would incorrectly expand the abbreviation. I've fixed that by also taking into account  the char, if any, that precedes the current insert buffer.